### ISSUE TITLE
The embedded, web3.js doesn't fully support interaction with autonity contract

### ISSUE THREAD

**Issue by piersy** [clearmatics/autonity#483]
# Problem

See the error returned when trying to execute `getComittee()` from the geth console.
```
> autonity.getCommittee()
Error: invalid solidity type!: tuple[]
```

The problem is discussed [here](https://ethereum.stackexchange.com/questions/36229/invalid-solidity-type-tuple) with the suggested solution being to upgrade web3.js to 1.0.0-beta.36.

# Implementation

Replace `internal/jsre/deps/web3.js` with a more up to date web3.js

# Acceptance Criteria

We should not be restricted because of `web3.js` in our interaction with the autonity contract via the geth console.

**Comment by piersy** [comment]
I have had a shot at doing this but I cannot get this working.

I got the mentioned [release](https://github.com/ethereum/web3.js/releases/tag/v1.0.0-beta.36) and unpacked it and ran.
```
npm install
npm run-script build
```
Then copied the built `web3.js` to `internal/jsre/deps/web3.js` then moved to the autonity project and ran 
 ```
go generate internal/jsre/deps/deps.go
make all
```
Then moved to the `local-autonity-deployment-utility` and started a new network, but all attempts to run the geth console fail with.
```
$ ../autonity/build/bin/autonity attach ipc:./network-data/node3/data/autonity.ipc
Fatal: Failed to start the JavaScript console: web3.js: web3.js: Line 11847:23 Invalid regular expression: re2: Invalid \1 <backreference> (and 3 more errors)
```

There is apparently a node library called re2 and it [doesn't support backreferences] (https://github.com/uhop/node-re2#backreferences) (I don't know if this is the problem) I cannot find any reference to re2 inside the web3.js release. So without a lot more investigation this ticket is not going anywhere.

**Comment by yazzaoui** [comment]
After spending more time than I should have, I give up on this one for the time being.

With the latest Autonity upstream update, the console javascript runtime environment has been replaced by [goja](https://github.com/dop251/goja) which is ECMAScript 5.1 compliant. I was hoping that this runtime supported the [RE2](https://www.npmjs.com/package/re2) which was causing the issue.

On the upstream merge branch I've replaced the web3.js library with the [latest one available ](https://github.com/ethereum/web3.js/tree/1.x/dist) (1.2.7). After regenerating the bindings, I was greeted by the following error : 
```
This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.
Fatal: Failed to start the JavaScript console: web3.js: ReferenceError: Uint8Array is not defined at web3.js
```

I figured out that `buffer` in its 5.X version was a dependency of `web3-bzz` (swarm bindings) and `web3-ssh` (whisper bindings). Since we removed in our project Swarm and Whisper I decided to remove those packages from web3 as well. After spending a great amount of time, due to my ignorance of how package management is done in the frontend word (I ended up removing every `node_module` folder even in the sub-packages ones), I finally managed to build a version of web3 without those unused bindings. Well, it turned out that `buffer 5.x` was still being used by some other components.

I decided then to follow the approach that Piers found on [stackoverflow](https://stackoverflow.com/questions/15806152/how-do-i-override-nested-npm-dependency-versions). Running `npm shrinkrap` generated the `npm-shrinkrap.json` file which made explict all the dependencies required by the library. I replaced all occurrences of `Buffer 5.X` by `Buffer 4.9.2` and managed to build the library with the correct Buffer version, after a few hours fighting against the build tools.. Obviously, the `npm run-script build` command didn't work, the correct one was `npm install && gulp`. 
Learned that running `npm install` a second time will regenerate weirdly `npm-shrinkrap.json` with the `Buffer 5.x` version and that running `lerna bootstrap` will fail for some unknown reasons. Obviously the 
  
After that, I recompiled Autonity, But I had a new error with unsupported `btoa` bindings. So one of the dependency was assuming that if the runtime wasn't node then the runtime had dom support (a browser). Hacked the code to fix that. Recompiled Autonity. Had another similar issue in another package. Fixed it. Another one again. Fixed it.

New error message : `Failed to start the JavaScript console: web3.js: ReferenceError: Uint8Array is not defined` 
I give up.
